packages/repo/src/mst/mst.ts

```diff
@@ -755,17 +774,33 @@ export class MST {
     return cids
   }
 
-  async addBlocksForPath(key: string, blocks: BlockMap) {
-    const serialized = await this.serialize()
-    blocks.set(serialized.cid, serialized.bytes)
-    const index = await this.findGtOrEqualLeafIndex(key)
-    const found = await this.atIndex(index)
-    if (found && found.isLeaf() && found.key === key) {
-      return
+  // A covering proof is all MST nodes (leaves excluded) needed to prove the value of a given leaf
+  // and its siblings to its immediate right and left (if applicable)
+  // We simply find the immediately preceeding node and then walk from that node until we reach the
+  // first key that is greater than the requested key (the right sibling)
+  async getCoveringProof(key: string): Promise<BlockMap> {
+    const leftSibling = await this.preceedingKey(key)
+    const blocks = new BlockMap()
+    for await (const node of this.walkFrom(leftSibling)) {
+      if (node.isTree()) {
+        const serialized = await node.serialize()
+        blocks.set(serialized.cid, serialized.bytes)
+      } else if (node.key > key) {
+        break
+      }
     }
+    return blocks
+  }
+
+  async preceedingKey(key: string): Promise<string> {
+    const index = await this.findGtOrEqualLeafIndex(key)
     const prev = await this.atIndex(index - 1)
-    if (prev && prev.isTree()) {
-      await prev.addBlocksForPath(key, blocks)
+    if (!prev) {
+      return ''
```

Comment:
Just want to double-check that this can only happen on the leftmost key and not e.g. in some bad case like a missing `key`.  From reading it seems that is the case, just looking for a double-check.  (For example, one unintended result could be that `getCoveringProof()` ends-up being very large by starting from the left edge of the tree in a bad case.)

Reply:
Yup that's right 👍